core/src/main/scala/org/apache/spark/rpc/RpcEnv.scala

```diff
@@ -264,9 +264,15 @@ object RpcTimeout {
     require(timeoutPropList.nonEmpty)
 
     // Find the first set property or use the default value with the first property
-    val foundProp = timeoutPropList.view.map(x => (x, conf.getOption(x))).filter(_._2.isDefined).
-      map(y => (y._1, y._2.get)).headOption.getOrElse(timeoutPropList.head, defaultValue)
-
+    val itr = timeoutPropList.iterator
+    var foundProp =  (timeoutPropList.head,defaultValue)
+    while (itr.hasNext && (foundProp == (timeoutPropList.head,defaultValue))){
```

Comment:
It's up to you.  For me, I think the loop might be a little cleaner by using an Option like this:

     var foundProp =  None: Option[(String, String)]
     while (itr.hasNext && foundProp.isEmpty){
       ...

Then get the final result with `foundProp.getOrElse(timeoutPropList.head, defaultValue)`.  It's probably not a big deal so if you prefer the way it is now, I can merge that.